### DISCUSSION TITLE
Ack problem after reconnection

### DISCUSSION THREAD

**Discussion by Ryuuji159** [socketio/socket.io#4321] [Q&A]
When a client reconnects while the server is waiting for an acknowledgement and the client, once reconnected, executes the old ack, it appears that the server isn't receiving it.

Is that an edge case that I have to take care of myself? or should socket.io be sorting that out and my code is the problem?

**Comment by darrachequesne** [comment]
Hi! I'm not able to reproduce the issue you are experiencing:

```js
io.use((socket, next) => {
  setTimeout(next, 10000);
});

io.on("connection", (socket) => {
  console.log(`connect ${socket.id}`);
});
```

When the client reconnects before the given delay, the middleware is properly called again. Or am I missing something?
